pkg/sql/fingerprint_span.go

```diff
@@ -163,12 +177,14 @@ func (p *planner) fingerprintSpanFanout(
 				rv.fingerprint = rv.fingerprint ^ localFingerprint
 				rv.Unlock()
 			}
-			return nil
 		})
 	}
+
+	close(spanChannel)
 	if err := grp.Wait(); err != nil {
 		return 0, nil, err
 	}
+	// TODO: Why lock and unlock if all threads done?
```

Comment:
@stevendanna can you explain why you lock and unlock before returning?

Reply:
If I had to guess, we massaged this code a few times during development and ended up with oddity as a result.  If you do remove this, make sure to run the tests under --race just in case. But, I agree we don't need to lock here.